lib/src/main/java/tech/relaycorp/relaydroid/ThirdPartyEndpoint.kt

```diff
@@ -1,5 +1,109 @@
 package tech.relaycorp.relaydroid
 
-public sealed class ThirdPartyEndpoint(override val address: String) : Endpoint
-public class PrivateThirdPartyEndpoint(address: String) : ThirdPartyEndpoint(address)
-public class PublicThirdPartyEndpoint(address: String) : ThirdPartyEndpoint(address)
+import org.bouncycastle.asn1.x500.style.BCStyle
+import tech.relaycorp.relaydroid.storage.persistence.PersistenceException
+import tech.relaycorp.relaynet.RelaynetException
+import tech.relaycorp.relaynet.issueDeliveryAuthorization
+import tech.relaycorp.relaynet.wrappers.x509.Certificate
+import tech.relaycorp.relaynet.wrappers.x509.CertificateException
+import java.security.PublicKey
+import java.time.ZonedDateTime
+
+public sealed class ThirdPartyEndpoint(
+    override val thirdPartyAddress: String
+) : Endpoint {
+
+    public companion object {
+
+        // Private
+
+        @Throws(PersistenceException::class)
+        public suspend fun loadPrivate(
+            firstPartyAddress: String, thirdPartyAddress: String
+        ): PrivateThirdPartyEndpoint? {
+            val key = "${firstPartyAddress}_$thirdPartyAddress"
+            return Storage.privateThirdPartyAuthorization.get(key)?.let {
+                PrivateThirdPartyEndpoint(firstPartyAddress, thirdPartyAddress, it)
+            }
+        }
+
+        @Throws(
+            PersistenceException::class,
+            InvalidFirstPartyEndpointAddressException::class
+        )
+        public suspend fun importPrivateAuthorization(pda: Certificate): PrivateThirdPartyEndpoint {
```

Comment:
I'd suggest to remove the word "private" because there's nothing called "private authorization". `importAuthorization()` would also be consistent with `issueAuthorization`

I think this method too belongs to `PrivateThirdPartyEndpoint`.